packages/components/src/components/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,194 @@
+};
+
+const TabConfigContext = createContext<{
+    value: string;
+    disabled: boolean;
+}>({
+    value: '',
+    disabled: false,
+});
+
+const TabTriggerContext = createContext<{
+    addTrigger: (trigger: TabTrigger) => void;
+}>({
+    addTrigger: () => {},
+});
+
+export const TabsRoot = (
+    {
+        children,
+        activeTab: propsActiveTab,
+        defaultActiveTab,
+        onActiveTabChange,
+        size = 'default',
+        ...props
+    }: TabsRootProps,
+    ref: ForwardedRef<HTMLDivElement>,
+) => {
+    const [activeTab, setActiveTab] = useControllableState({
+        prop: propsActiveTab,
+        defaultProp: defaultActiveTab,
+        onChange: onActiveTabChange,
+    });
+
+    const { triggerListRef, activeIndicatorRef, triggers, triggersOutOfView, addTrigger } = useTabTriggers({
+        activeTab,
+    });
+
+    return (
+        <TabTriggerContext.Provider value={{ addTrigger }}>
+            <RadixTabs.Root
+                ref={ref}
+                className={styles.root}
+                onValueChange={(value) => {
+                    if (value) {
+                        setActiveTab(value);
+                    }
+                }}
+                value={activeTab ?? triggers[0]?.value}
+                {...props}
+            >
+                <div className={styles.triggerListWrapper}>
+                    <RadixTabs.List ref={triggerListRef} data-size={size} className={styles.triggerList}>
+                        {triggers.map((trigger) => (
+                            <RadixTabs.Trigger
+                                key={trigger.value}
+                                value={trigger.value}
+                                disabled={trigger.disabled}
+                                className={styles.trigger}
+                                asChild
+                            >
+                                {trigger.element}
+                            </RadixTabs.Trigger>
+                        ))}
+                    </RadixTabs.List>
+                    <Dropdown.Root>
+                        {triggersOutOfView.length > 0 && (
+                            <Dropdown.Trigger data-test-id="overflow-items-dropdown-trigger">
+                                <Button emphasis="default" aspect="square" size="small">
+                                    <IconDotsHorizontal size={16} />
+                                </Button>
+                            </Dropdown.Trigger>
+                        )}
+                        <Dropdown.Content align="end" data-test-id="overflow-items-dropdown-content">
+                            {triggersOutOfView.map((trigger) => (
+                                <Dropdown.Item
+                                    disabled={trigger.disabled}
+                                    onSelect={() => {
+                                        setActiveTab(trigger.value);
+                                    }}
```

Comment:
As it is in `.map`, move it out and add a `useCallback`

Reply:
good catch!